### ISSUE TITLE
feature_request(update): pylint and dependencies

### ISSUE THREAD

**Issue by Kristinita** [thusoy/grunt-pylint#20]
### 1. Summary

It would be nice, if pylint and it dependencies will be updated to the latest versions.

grunt-pylint doesn't work for me for simple scripts.

#### 2.1. Files

+ [**Example configuration on KiraPylint branch of my demo repository**](https://github.com/Kristinita/SashaGruntDebugging/tree/KiraPylint)

+ `Gruntfile.coffee`:

```coffee
module.exports = (grunt) ->
    grunt.initConfig
        pylint:
            options:
                externalPylint: false
            dist:
                src: ["kira_grunt_pylint.py"]

    grunt.loadNpmTasks 'grunt-pylint'
    return

```

+ `kira_grunt_pylint.py`:

```python
"""Docstrings."""
import ruamel.yaml as yaml
YAMLCONFIG = yaml.safe_load(open('site_variables.yaml', encoding='utf-8'))
print(YAMLCONFIG)

```

#### 2.2. Expected behavior

+ [**Travis CI demonstration**](https://travis-ci.org/Kristinita/SashaGruntDebugging/builds/506618309#L261-L272)

If `externalPylint: true` (install via `pip install pylint`):

```python
Running "pylint:dist" (pylint) task
Verifying property pylint.dist exists in config…OK
Files: kira_grunt_pylint.py
Options: errorsOnly=false, force=false, outputFormat="colorized", messageTemplate="short", report=false, externalPylint
Executing with python at: python, and arguments: -c import pylint; pylint.run_pylint() --msg-template="line {line}: {msg} ({symbol})" --output-format=colorized --reports=n kira_grunt_pylint.py
------------------------------------
Your code has been rated at 10.00/10
>> No lint in kira_grunt_pylint.py
Done.
The command "grunt pylint --verbose" exited with 0.
```

#### 2.3. Actual behavior

+ [**Travis CI demonstration**](https://travis-ci.org/Kristinita/SashaGruntDebugging/builds/506617809#L261-L308)

Else `externalPylint: false`:

```python
Running "pylint:dist" (pylint) task
Verifying property pylint.dist exists in config…OK
Files: kira_grunt_pylint.py
Options: errorsOnly=false, force=false, outputFormat="colorized", messageTemplate="short", report=false, externalPylint=false
Executing with python at: python, and arguments: -c import sys; sys.path.insert(0, r"/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib"); import pylint; pylint.run_pylint() --msg-template="line {line}: {msg} ({symbol})" --output-format=colorized --reports=n kira_grunt_pylint.py
Stderr from pylint: No config file found, using default configuration
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/pylint/__init__.py", line 23, in run_pylint
    Run(sys.argv[1:])
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/pylint/lint.py", line 1313, in _init_
    linter.check(args)
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/pylint/lint.py", line 734, in check
    self._do_check(files_or_modules)
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/pylint/lint.py", line 865, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/pylint/lint.py", line 945, in check_astroid_module
    walker.walk(ast_node)
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/pylint/utils.py", line 938, in walk
    self.walk(child)
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/pylint/utils.py", line 935, in walk
    cb(astroid)
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/pylint/checkers/variables.py", line 1004, in visit_import
    module = next(node.infer_name_module(parts[0]))
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/astroid/context.py", line 49, in cache_generator
    for result in generator:
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/astroid/bases.py", line 302, in wrapped
    for res in _func(node, context, **kwargs):
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/astroid/inference.py", line 123, in infer_import
    yield self.do_import_module(name)
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/astroid/mixins.py", line 129, in do_import_module
    relative_only=level and level >= 1)
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/astroid/scoped_nodes.py", line 492, in import_module
    return MANAGER.ast_from_module_name(absmodname)
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/astroid/manager.py", line 129, in ast_from_module_name
    return self.ast_from_module(module, modname)
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/astroid/manager.py", line 193, in ast_from_module
    if modutils.is_python_source(filepath):
  File "/home/travis/build/Kristinita/SashaGruntDebugging/node_modules/grunt-pylint/tasks/lib/astroid/modutils.py", line 492, in is_python_source
    return os.path.splitext(filename)[1][1:] in PY_SOURCE_EXTS
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/posixpath.py", line 122, in splitext
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
Warning: Task "pylint:dist" failed. Use --force to continue.
Aborted due to warnings.
The command "grunt pylint --verbose" exited with 3.
```

### 3. Recommendation

It would be highly nice, if grunt-pylint will use latest pylint and it dependencies versions, not specific [**as at the time**](https://github.com/thusoy/grunt-pylint/blob/4a5a765be5c01845273c24976c92c37d122dfc4a/package.json#L23).

Python and Pylint — maintained projects. In new versions:

1. New features implemented
1. Bugs fixed

Previous pylint versions show errors for correct code and haven't new features. We need often update pylint and dependencies for grunt-pylint. Always the latest dependencies versions solve this problem.

Thanks.

**Comment by thusoy** [comment]
Hi! I'd be happy to accept a PR to update the deps, otherwise I might get around to it next weekend.

**Comment by thusoy** [comment]
Hi! This should now be fixed in version 2.0.1.
